src/schema.c

```diff
@@ -113,14 +135,84 @@ ldap_schema_object_classes(const ldap_schema_t *schema)
 LDAPAttributeType**
 ldap_schema_attribute_types(const ldap_schema_t* schema)
 {
-    if (!schema)
+    if (!schema || !schema->attribute_types_by_oid)
```

Comment:
Consider separating checks of `schema` and `attribute_types_by_oid`. If we have separate checks for each case, we can accurately say what exactly the error is.